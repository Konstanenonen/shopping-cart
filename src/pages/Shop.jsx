import React, { useState } from 'react';
import ShopCartBar from '../components/ShopCartBar';
import ShopItem from '../components/ShopItem';
import shopItems from '../data/shopItems';
import styles from '../styles/pages/Shop.module.scss';

function Shop() {
  const [itemsAmount, setItemsAmount] = useState(0);
  const items = shopItems;

  const addToCart = (amount) => {
    setItemsAmount((i) => i + amount);
  };

  const itemElements = items.map((item) => (
    <ShopItem
      key={item.title}
      title={item.title}
      description={item.description}
      addToCart={addToCart}
    />
  ));

  return (
    <>
      <ShopCartBar itemsAmount={itemsAmount} />
      <div className={styles.container}>{itemElements}</div>
    </>
  );
}

export default Shop;
