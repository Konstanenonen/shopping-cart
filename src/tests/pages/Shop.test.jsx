/* global it describe expect jest */

import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';
import Shop from '../../pages/Shop';

describe('Shop page', () => {
  it('Renders shop item data to ShopItem components', () => {
    const { container } = render(<Shop />);
    expect(container).toMatchSnapshot();
  });

  it('ShopCartBar has correct amount of items after addition', () => {
    render(<Shop />);
    const input = screen.getAllByRole('spinbutton')[0];
    const button = screen.getAllByRole('button', { name: 'Add to cart' })[0];
    const shoppingCart = screen.getByText(/items/i);

    act(() => {
      userEvent.type(input, '12');
      userEvent.click(button);
    });

    expect(shoppingCart.innerHTML).toBe('Items in shopping cart: 12');
  });
});
