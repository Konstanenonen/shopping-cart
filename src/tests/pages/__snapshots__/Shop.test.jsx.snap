// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop page Renders shop item data to ShopItem components 1`] = `
<div>
  <div
    class="container"
  >
    <p>
      Items in shopping cart: 
      0
    </p>
    <button
      class="button"
      type="button"
    >
      CHECKOUT
    </button>
  </div>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <h3
        class="title"
      >
        Item title1
      </h3>
      <p
        class="text"
      >
        Item description
      </p>
      <div
        class="row"
      >
        <button
          class="button"
          type="button"
        >
          -1
        </button>
        <label
          class="label"
          for="item-amount"
        >
          Add more to order:
          <input
            id="item-amount"
            type="number"
            value="0"
          />
        </label>
        <button
          class="button"
          type="button"
        >
          +1
        </button>
      </div>
      <button
        class="button"
        type="button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="container"
    >
      <h3
        class="title"
      >
        Item title2
      </h3>
      <p
        class="text"
      >
        Item description
      </p>
      <div
        class="row"
      >
        <button
          class="button"
          type="button"
        >
          -1
        </button>
        <label
          class="label"
          for="item-amount"
        >
          Add more to order:
          <input
            id="item-amount"
            type="number"
            value="0"
          />
        </label>
        <button
          class="button"
          type="button"
        >
          +1
        </button>
      </div>
      <button
        class="button"
        type="button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="container"
    >
      <h3
        class="title"
      >
        Item title3
      </h3>
      <p
        class="text"
      >
        Item description
      </p>
      <div
        class="row"
      >
        <button
          class="button"
          type="button"
        >
          -1
        </button>
        <label
          class="label"
          for="item-amount"
        >
          Add more to order:
          <input
            id="item-amount"
            type="number"
            value="0"
          />
        </label>
        <button
          class="button"
          type="button"
        >
          +1
        </button>
      </div>
      <button
        class="button"
        type="button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="container"
    >
      <h3
        class="title"
      >
        Item title4
      </h3>
      <p
        class="text"
      >
        Item description
      </p>
      <div
        class="row"
      >
        <button
          class="button"
          type="button"
        >
          -1
        </button>
        <label
          class="label"
          for="item-amount"
        >
          Add more to order:
          <input
            id="item-amount"
            type="number"
            value="0"
          />
        </label>
        <button
          class="button"
          type="button"
        >
          +1
        </button>
      </div>
      <button
        class="button"
        type="button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="container"
    >
      <h3
        class="title"
      >
        Item title5
      </h3>
      <p
        class="text"
      >
        Item description
      </p>
      <div
        class="row"
      >
        <button
          class="button"
          type="button"
        >
          -1
        </button>
        <label
          class="label"
          for="item-amount"
        >
          Add more to order:
          <input
            id="item-amount"
            type="number"
            value="0"
          />
        </label>
        <button
          class="button"
          type="button"
        >
          +1
        </button>
      </div>
      <button
        class="button"
        type="button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="container"
    >
      <h3
        class="title"
      >
        Item title6
      </h3>
      <p
        class="text"
      >
        Item description
      </p>
      <div
        class="row"
      >
        <button
          class="button"
          type="button"
        >
          -1
        </button>
        <label
          class="label"
          for="item-amount"
        >
          Add more to order:
          <input
            id="item-amount"
            type="number"
            value="0"
          />
        </label>
        <button
          class="button"
          type="button"
        >
          +1
        </button>
      </div>
      <button
        class="button"
        type="button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="container"
    >
      <h3
        class="title"
      >
        Item title7
      </h3>
      <p
        class="text"
      >
        Item description
      </p>
      <div
        class="row"
      >
        <button
          class="button"
          type="button"
        >
          -1
        </button>
        <label
          class="label"
          for="item-amount"
        >
          Add more to order:
          <input
            id="item-amount"
            type="number"
            value="0"
          />
        </label>
        <button
          class="button"
          type="button"
        >
          +1
        </button>
      </div>
      <button
        class="button"
        type="button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="container"
    >
      <h3
        class="title"
      >
        Item title8
      </h3>
      <p
        class="text"
      >
        Item description
      </p>
      <div
        class="row"
      >
        <button
          class="button"
          type="button"
        >
          -1
        </button>
        <label
          class="label"
          for="item-amount"
        >
          Add more to order:
          <input
            id="item-amount"
            type="number"
            value="0"
          />
        </label>
        <button
          class="button"
          type="button"
        >
          +1
        </button>
      </div>
      <button
        class="button"
        type="button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="container"
    >
      <h3
        class="title"
      >
        Item title9
      </h3>
      <p
        class="text"
      >
        Item description
      </p>
      <div
        class="row"
      >
        <button
          class="button"
          type="button"
        >
          -1
        </button>
        <label
          class="label"
          for="item-amount"
        >
          Add more to order:
          <input
            id="item-amount"
            type="number"
            value="0"
          />
        </label>
        <button
          class="button"
          type="button"
        >
          +1
        </button>
      </div>
      <button
        class="button"
        type="button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="container"
    >
      <h3
        class="title"
      >
        Item title10
      </h3>
      <p
        class="text"
      >
        Item description
      </p>
      <div
        class="row"
      >
        <button
          class="button"
          type="button"
        >
          -1
        </button>
        <label
          class="label"
          for="item-amount"
        >
          Add more to order:
          <input
            id="item-amount"
            type="number"
            value="0"
          />
        </label>
        <button
          class="button"
          type="button"
        >
          +1
        </button>
      </div>
      <button
        class="button"
        type="button"
      >
        Add to cart
      </button>
    </div>
  </div>
</div>
`;
